### PR TITLE
2.8 Adding tests to verify the crd generation fix

### PR DESCRIPTION
**Pull request by anupama2501**
Forwardport: https://github.com/rancher/rancher/pull/42149

### CODE REVIEW

**Review by caliskanugur — APPROVED** [review]
LGTM

**Review by igomez06 — APPROVED** [review]
LGTM

File: tests/framework/extensions/kubectl/explain.go [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+	apisV1 "github.com/rancher/rancher/pkg/apis/provisioning.cattle.io/v1"
+	"github.com/rancher/rancher/tests/framework/clients/rancher"
+	"github.com/rancher/rancher/tests/framework/extensions/kubeconfig"
+	"github.com/rancher/rancher/tests/framework/extensions/workloads"
+	"github.com/rancher/rancher/tests/framework/extensions/workloads/pods"
+	corev1 "k8s.io/api/core/v1"
+)
+
+// Explain is a helper function that creates a Job by calling the helper CreateJobAndRunKubectlCommands and executing kubectl explain in the pod of the Job
+//returns the output from the pod logs.
```
Comment (igomez06) [inline tests/framework/extensions/kubectl/explain.go:16]: need to fix this lint mistake, you need a space here.

Comment (igomez06) [reply tests/framework/extensions/kubectl/explain.go:16]: Also public_api is not a proper directory name. It should be publicapi as one name

**Review by igomez06 — COMMENTED** [review]
Two lint mistakes, and we're good to go

**Review by igomez06 — APPROVED** [review]
LGTM

**Pull request merged**